agent_test.go

```diff
@@ -39,3 +41,38 @@ func TestMain(m *testing.M) {
 	os.Exit(ret)
 
 }
+
+func TestParseKernelCmdline(t *testing.T) {
+	type args struct {
+		cmdline string
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{"Empty cmdline",
+			args{""}, false},
+		{"Log config info",
+			args{"agent.debug=info"}, false},
+		{"Log config debug",
+			args{"agent.debug=debug "}, false},
+		{"Log config warn",
+			args{" systemd kernel=someting agent.debug=warn "}, false},
+		{"Log config empty with space",
+			args{"agent.debug= debug "}, false},
+		{"Log config empty",
+			args{"agent.debug="}, false},
+	}
```

Comment:
You could also test in case a wrong parameter is given. Something like:
`agent.debug=ino`
`agent.debug=wran`
`agent.debug=degbu`